roles/tests-integration/tests/pool_integration.rs

```diff
@@ -44,3 +48,32 @@ async fn success_pool_template_provider_connection() {
     assert_tp_message!(&sniffer.next_message_from_upstream(), NewTemplate);
     assert_tp_message!(sniffer.next_message_from_upstream(), SetNewPrevHash);
 }
+
+#[tokio::test]
+async fn test_sniffer_interrupter() {
+    let sniffer_addr = common::get_available_address();
+    let tp_addr = common::get_available_address();
+    let pool_addr = common::get_available_address();
+    let _tp = common::start_template_provider(tp_addr.port()).await;
+    use const_sv2::MESSAGE_TYPE_SETUP_CONNECTION_SUCCESS;
+    let message =
+        PoolMessages::Common(CommonMessages::SetupConnectionError(SetupConnectionError {
+            flags: 0,
+            error_code: "unsupported-feature-flags"
+                .to_string()
+                .into_bytes()
+                .try_into()
+                .unwrap(),
+        }));
+    let interrupt_msgs = InterruptMessage::new(
+        MessageDirection::ToDownstream,
+        MESSAGE_TYPE_SETUP_CONNECTION_SUCCESS,
+        message,
+        MESSAGE_TYPE_SETUP_CONNECTION_ERROR,
+        true,
+    );
+    let sniffer = common::start_sniffer(sniffer_addr, tp_addr, Some(vec![interrupt_msgs])).await;
+    let _ = common::start_pool(Some(pool_addr), Some(sniffer_addr)).await;
+    assert_common_message!(&sniffer.next_downstream_message(), SetupConnection);
+    assert_common_message!(&sniffer.next_upstream_message(), SetupConnectionError);
```

Comment:
hmm I guess this test gives us confidence that we triggered the "interrupt" system on sniffer, but how do we know that pool actually received the expected modified message?

should we assert the pool state afterwards?

or maybe use an extra sniffer?